### PR TITLE
Import Document

### PR DESCRIPTION
**Pull request by Pineirin**
Better implementation (from a memory, performance and code clarity point of view) of the pull request https://github.com/CERNDocumentServer/cds-ils/pull/619.

### CODE REVIEW

File: ui/src/importer/importerTaskDetails/ImportedDocuments.js [hunk 1]
```diff
@@ -43,6 +81,10 @@ export class ImportedDocuments extends React.Component {
     this.checkForData(taskId);
   }
 
+  componentDidUpdate() {
+    this.filteredRecords();
```
Comment (zzacharo) [inline ui/src/importer/importerTaskDetails/ImportedDocuments.js:85]: Isn't the `this.filteredRecords()` updating the state? If so, then I think you would be in a infinite loop of updates no?

Comment (Pineirin) [reply ui/src/importer/importerTaskDetails/ImportedDocuments.js:85]: Yes, an oversight from my part, I will fix it.

File: ui/src/importer/importerTaskDetails/ImportedDocuments.js [hunk 2]
```diff
@@ -61,15 +99,107 @@ export class ImportedDocuments extends React.Component {
...
+  };
+
+  calculateStatistics = data => {
+    const { statistics } = this.state;
+    let stats = _cloneDeep(statistics);
+    const records = data.records;
+    const newRecords = records.slice(this.lastIndex, records.length);
+
+    this.lastIndex += newRecords.length;
+    stats.records.value = data.loaded_entries + '/' + data.entries_count;
```
Comment (kprzerwa) [inline ui/src/importer/importerTaskDetails/ImportedDocuments.js:188]:
```suggestion
    stats.records.value = `${data.loaded_entries}/${data.entries_count}`;
```
addition on strings and numbers in javascript sometimes does not end well :)

File: ui/src/importer/importerTaskDetails/ImportedDocuments.js [hunk 3]
```diff
@@ -83,177 +213,171 @@ export class ImportedDocuments extends React.Component {
...
-        <JsonViewModal />
-        <SeriesImportDetailsModal />
-        <EitemImportDetailsModal />
-      </>
+  searchBarFilter = filteredRecords => {
+    const { searchText } = this.state;
+    let deb = debounce(searchText => {
+      const searchRecords = filteredRecords.filter(
+        record =>
+          record.entry_recid.includes(searchText) ||
```
Comment (kprzerwa) [inline ui/src/importer/importerTaskDetails/ImportedDocuments.js:266]:
```suggestion
          record.entry_recid?.includes(searchText) ||
```

File: ui/src/importer/importerTaskDetails/ImportedDocuments.js [hunk 4]
```diff
@@ -83,177 +213,171 @@ export class ImportedDocuments extends React.Component {
...
+          record.output_pid?.includes(searchText) ||
+          record.document?.title.includes(searchText) ||
+          record.eitem?.output_pid?.includes(searchText) ||
+          record.series?.filter(serie => serie.output_pid?.includes(searchText))
+            .length > 0
+      );
+      console.log('Tick');
```
Comment (kprzerwa) [inline ui/src/importer/importerTaskDetails/ImportedDocuments.js:273]: 👀

File: ui/src/importer/importerTaskDetails/ImportedDocuments.js [hunk 5]
```diff
@@ -83,177 +213,171 @@ export class ImportedDocuments extends React.Component {
...
+                    <Grid.Column width={3}>
+                      <Segment>
+                        Mode: {this.mode}
+                        <Divider />
+                        Status: {this.status}
+                      </Segment>
+                    </Grid.Column>
+                    <Grid.Column width={13}>
+                      <Statistic.Group widths="seven">
+                        {Object.keys(statistics).map(function(
```
Comment (kprzerwa) [inline ui/src/importer/importerTaskDetails/ImportedDocuments.js:335]: could we improve the readability of this part ? maybe makes sense to have a function component for this?

**Pull request merged**